ccm_web/client/src/hooks/useGlobals.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react'
+
+import usePromise from './usePromise'
+import * as api from '../api'
+import { Globals } from '../models'
+
+/* Hook for fetching global data and checking whether user is authenticated. */
+function useGlobals (ltiKey: string | undefined): [
+  Globals | undefined,
+  boolean | undefined,
+  boolean,
+  Error | undefined
+] {
+  const [globals, setGlobals] = useState(undefined as Globals | undefined)
+
+  const getGlobals = async (): Promise<Globals> => await api.getGlobals(ltiKey)
+  const [doGetGlobals, getGlobalsLoading, getGlobalsError] = usePromise(
+    getGlobals,
+    (value: Globals) => setGlobals(value)
+  )
+  useEffect(() => {
+    /* eslint-disable @typescript-eslint/no-floating-promises */
+    doGetGlobals()
+  }, [])
+
+  let isAuthenticated
+  if (globals !== undefined) {
+    if (ltiKey !== undefined && globals.useLTI) {
+      isAuthenticated = true
+    } else if (
+      globals.environment === 'development' &&
+      ltiKey === undefined &&
+      !globals.useLTI
+    ) {
+      isAuthenticated = true
+    }
+  } else if (getGlobalsError !== undefined) {
+    isAuthenticated = false
+  }
```

Comment:
I don't actually know how good this `useGlobals` approach is, but I'll attempt to explain it. 

First, for some reason, `ltijs` is not locking down the application route if you don't have an LTI Key (it is for the other routes, so not sure what's going on there. Maybe it needs to be accessible for the launch?). So we need some way of locking down the landing page to unauthenticated users. Hence, the `/api/globals` call. MyLA's behavior is kinda similar, but it uses `globals` from the `index` template.

I would like to allow developers to still use the application without LTI, but I'm not sure adding an alternate authentication scheme is necessary. To that end, I'm trying to use some `Globals` data to allow users to be seen as authenticated if it's a development environment (they'll be able to access Globals data because on that path, the `api` routes will be unprotected). I don't think this is risky from a security perspective, but I could be missing something.

Reply:
> I would like to allow developers to still use the application without LTI, 

With out the LTI how will you know what course, who the user is, what is user role? Based on that we are showing the feature in the landing page. If he has to merge sections then how will we know what courses to merge and from which term. We get the data from Launch information and allow him to do necessary feature action

Reply:
I understand that, and perhaps we can only take this idea so far.

One idea is that we could allow developers to configure a course and username (that actually exists), which would be sent over through `Globals` (this would be in a future PR). In the case of a real LTI integration, the `Globals` object could include the real user, course, and roles from the LTI launch.

Reply:
My thought, which may or may not make sense here, would be an interface that the contains all the relevant properties that lti provides.  We could then have something like an LTIIdentityProfvider when using LTI and a EnvVarIdentityProvider or ConfigFileIdentityProfier or something when using development (or testing?) if desired.  Perhaps someone else might want some other sort of IdentityProvider to integrate with their platform (unlikely)

Not even sure if Identity is the correct term here, I'm not sure what LTI is providing.

Reply:
@ssciolla I appreciate your enthusiasm for providing an alternative approach without LTI. But I feel we should park that idea for now - my 2 cents.

Reply:
Okay, I can rewind some of this and adjust `AppHandler`.

Reply:
@chrisrrowland, something like you what you are describing is what I had mind, but I guess we can postpone or make it a separate issue. In the meantime, if developing the UI under LTI constraints bothers you enough, we can work on getting `storybook` set up.

Reply:
@ssciolla It doesn't bother me.  It adds a minor level of annoyance but I can survive.

That said I understand wanting to shelf that as a nice to have but It doesn't seem like it really adds much effort to create the interface and the LTI implementation and leave any alternatives for a later effort.  

I am not in the weeds on this, though, so it could certainly be  more difficult than I'm envisioning lol

Reply:
Thanks for the additional thoughts, @chrisrrowland. I want to think about this a little more and separate it from this PR. I'm not really sure about my current approach; it would be better if the non-LTI alternative also provided some kind of key for authenticating to API endpoints. That definitely would make this more complicated (how much so, not sure). If any of my current approach is use-able in the future, it'll just be a copy and paste job from these commits.

Reply:
@pushyamig, see https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/40/commits/464fd0ccf98c3af0ad0112b2aa6e90c6d1e3841b

I would still like to know if others think my approach to locking down `/` makes sense. See `useGlobals` and `App.tsx`.

Reply:
The look down approach look ok. I don't see any concerns there. 
I am still trying to wrap my head around the promise/async/await with fetch. This looks bit complicated just to drilling downs what's happening in the `usePromise`. But at the high level i understand what it is doing

Reply:
Yeah, I don't have a mastery of it either. `async`, `await,` and `Promise` are all still pretty confusing to me; I plan to try to learn more about them as time allows.

Reply:
Marking this thread as resolved. Ask more questions if you see other issues related to this stuff.